litestar/utils/predicates.py

```diff
@@ -39,6 +39,11 @@
 from litestar.utils.helpers import unwrap_partial
 from litestar.utils.typing import get_origin_or_inner_type
 
+if sys.version_info >= (3, 10):
+    from inspect import iscoroutinefunction
+else:  # pragma: no cover
+    from asyncio import iscoroutinefunction
```

Comment:
```suggestion
    # In Python 3.9, AsyncMock is not detected
    # as a coroutine function, so one test was failing.
    from asyncio import iscoroutinefunction
```

Sorry, I missed this suggestion the first time :(